### PR TITLE
Print version and sha

### PR DESCRIPTION
**Pull request by jlb6740**
This updates adds a hash of most the files used for building and running wasmscore in order to check that has again inside the container at runtime. This way we can check if the stated build version is different from the version actually running.

### CODE REVIEW

**Comment by jlb6740** [comment]
A tricky part here will be updating this version number with included sha programmatically during CI checks. For now this is expected to be done by hand.

**Comment by rahulchaphalkar** [comment]
When I `./build.sh`, I get a sha mismatch. I've just checked out the branch, no changes to files.
```
/repos/wasm-score$ ./build.sh

Build information
-----------------
Major: 0
Minor: 1
Revision: 1
Build Sha: 54bbb97 vs 535a7b1 (calculated)

Building wasmscore-v0.1.1.54bbb97 for x86_64.

```
It still builds successfully. When I go ahead and run the benchmark, I get a message -
```
/repos/wasm-score$ docker run -ti wasmscore

WasmScore
Build SHA: 54bbb97 vs 535a7b1 (calculated) (**run invalid**)

``` 
And it then runs to completion.

- Sha signatures should match in this case
- What is the expected behavior if they don't match due to any mods I have done? Simply print out the above messages, or affect the behavior in any other way (here or in WSF)?

**Comment by jlb6740** [comment]
> /repos/wasm-score$ docker run -ti wasmscore
> 
> WasmScore
> Build SHA: 54bbb97 vs 535a7b1 (calculated) (**run invalid**)

Thanks. I tried on a different machine and also had a different hash. Checking it out now.

**Comment by rahulchaphalkar** [comment]
@jlb6740 
I tried with latest changes, however I am still getting a sha mismatch. 
```
/wasm-score$ ./build.sh

Build information
-----------------
Major: 0
Minor: 1
Revision: 1
Build Sha: be2c738 vs 538cb6b (calculated)

Building wasmscore-v0.1.1.be2c738 for x86_64.
```
I checked manually as well and my sha is different than what is included in `config.inc`.
I have attached the output of my sha calculations for reference, I think comparing it with similar output from your local system will help debug this. This outputs individual sha's, using this command:
```
find . -type f -name '*.wasm' | sort -n | xargs -I{} sha1sum add_time_metric.diff build.sh requirements.txt Dockerfile wasmscore.py {}
```
[individual_sha.txt](https://github.com/bytecodealliance/wasm-score/files/15066664/individual_sha.txt)

One thing we see from this is, sha for non-wasm files is being calculated over and over again for each wasm file. Should be ok for the sha itself, but you might consider calculating wasm shas first and store it in a temp file/var, and then non-wasm shas, to avoid this.

**Comment by jlb6740** [comment]
@rahulchaphalkar  ... Can you try again. I was in the middle of updating when you were checking. I needed to do a sort of the find as results from find were returned differently in and outside the container but turns that  even the sort -n was not consistent. stream2, stream was the order outside the container while stream, stream2 was the order inside. OS differences were Ubuntu 22.04 vs Ubuntu 20.04. I found a sort that had some consistency, "sort d" .

**Comment by rahulchaphalkar** [comment]
The build/run passes for me now with sha matching.

**Review by rahulchaphalkar — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -82,8 +82,9 @@ ADD benchmarks /sightglass/benchmarks
 
 # Copy driver/helpers into the image
 WORKDIR /
-COPY wasmscore.py /sightglass/wasmscore.py
 COPY wasmscore.sh /
+COPY wasmscore.py /sightglass/wasmscore.py
```
[the remark this answers is not in the document]
Comment (jlb6740) [reply Dockerfile:86]: One is a python script, which we don't call directly.

**Pull request merged**